guildwars2/guild/general.py

```diff
@@ -17,23 +17,43 @@ class GeneralGuild:
 
     @guild.command(name="info")
     @commands.cooldown(1, 20, BucketType.user)
-    async def guild_info(self, ctx, *, guild_name: str):
+    async def guild_info(self, ctx, *, guild_name=None):
         """General guild stats
 
         Required permissions: guilds
         """
+        # Read preferred guild from DB
+        guild_id = await self.get_preferred_guild(ctx.author)
+        # Get Guild name if ID already stored
+        if guild_id:
+            try:
+                endpoint_name = "guild/{0}".format(guild_id)
+                results = await self.call_api(endpoint_name)
+                guild_name = results["name"]
+            except APIError as e:
+                return await self.error_handler(ctx, e)
+        elif guild_name is not None:
+            try:
+                endpoint_id = "guild/search?name=" + guild_name.replace(
+                    ' ', '%20')
+                guild_id = await self.call_api(endpoint_id)
+                guild_id = guild_id[0]
+
+            except (IndexError, APINotFound):
+                return await ctx.send("Invalid guild name")
+            except APIError as e:
+                return await self.error_handler(ctx, e)
+        else:
+            return await self.bot.send_cmd_help(ctx)
```

Comment:
This whole chunk can be moved to a new function, since it's reused in other guild commands too

Reply:
Made into functions guildid_to_guildname and reverse